shared/src/utils/validators/is-string-literal-union.ts

```diff
@@ -0,0 +1,26 @@
+import type { ValidationOptions, ValidationArguments } from 'class-validator';
+import { StrictObject } from '../strict-object';
+import { makeValidator } from './make-validator';
+
+export function isStringLiteralUnion<T extends string>(
+    allowedValues: { [key in T]: true },
+    valueToBeValidated: unknown
+): boolean {
+    return (
+        typeof valueToBeValidated === 'string' &&
```

Comment:
I think this check is unnecessary.

Reply:
Wait - if the value is an object this would probably throw? So disregard this.

Reply:
No, this works:

```ts
const x = { a: 1 } as unknown as Record<any, any>

console.log(x['a'])
console.log(x[1])
console.log(x[{ w: 'i' } as unknown as string])
```

prints

```console
[LOG]: 1 
[LOG]: undefined 
[LOG]: undefined 
```

Tested with <https://www.typescriptlang.org/play> on TS 4.8.4.

Reply:
I'll still keep the check, though. Just to be safe, as it's not that calming to just trust a weird type wouldn't crash there.